src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingChannelExistenceChecker.java

```diff
@@ -0,0 +1,293 @@
+    this.clientSequencer = clientSequencer;
+  }
+
+  /**
+   * C'tor for StreamingChannelExistenceChecker
+   *
+   * @param tableName TableName for checking Channel Name in it.
+   * @param connectorConfig connectorConfig passed in SF Kafka Connect Configuration
+   */
+  public StreamingChannelExistenceChecker(String tableName, Map<String, String> connectorConfig) {
+    this(tableName, connectorConfig, 0L);
+  }
+
+  /**
+   * Determines if a Channel Exists or not for a table defined in this class's ctor.
+   *
+   * @param channelName Name of Channel to Check for Existence
+   * @return True if it exists, false if it doesn't.
+   */
+  public boolean checkChannelExistence(final String channelName) {
+    // Re-using client from IngestSDK since it handles lots of proxy props.
+    // Please don't close this client.
+    CloseableHttpClient httpClient = HttpUtil.getHttpClient(snowflakeURL.getAccount());
+    Object credential = generateCredentialObjectForStreamingRestApi();
+    RequestBuilder requestBuilder =
+        new RequestBuilder(
+            snowflakeURL,
+            connectorConfig.get(Utils.SF_USER),
+            credential,
+            httpClient,
+            /* This String is used for Telemetry */
+            String.format(
+                "%s_%s_%s",
+                STREAMING_CUSTOM_REST_API_CLIENT, channelName, System.currentTimeMillis()));
+
+    ChannelExistenceCheckerRequest channelExistenceCheckerRequest =
+        new ChannelExistenceCheckerRequest();
+    ChannelExistenceCheckerRequest.ChannelStatusRequestDTO requestDTO =
+        new ChannelExistenceCheckerRequest.ChannelStatusRequestDTO(
+            this.connectorConfig.get(Utils.SF_DATABASE),
+            this.connectorConfig.get(Utils.SF_SCHEMA),
+            tableName,
+            channelName,
+            this.clientSequencer);
+    channelExistenceCheckerRequest.setChannels(Collections.singletonList(requestDTO));
+    channelExistenceCheckerRequest.setRole(connectorConfig.get(Utils.SF_ROLE));
+    channelExistenceCheckerRequest.setRequestId(UUID.randomUUID().toString());
+
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+      String requestPayload = objectMapper.writeValueAsString(channelExistenceCheckerRequest);
+
+      HttpUriRequest request =
+          requestBuilder.generateStreamingIngestPostRequest(
+              requestPayload,
+              CHANNEL_STATUS_ENDPOINT,
+              DEBUG_PREFIX /* Message passed on to Ingest SDK */);
+
+      ChannelExistenceCheckerResponse channelExistenceCheckerResponse;
+      try (CloseableHttpResponse closeableHttpResponse = httpClient.execute(request)) {
+
+        if (closeableHttpResponse == null) {
+          LOGGER.warn(
+              "Null response obtained in {} for channel:{}", CHANNEL_STATUS_ENDPOINT, channelName);
```

Comment:
why we need a log if we're going to throw?

Reply:
to know which channel name had this issue.